### PR TITLE
Spinner on login

### PR DESCRIPTION
**Pull request by gabedealmeida**

### CODE REVIEW

File: ui/src/store/account.js [hunk 1]
```diff
@@ -53,6 +54,9 @@ export default {
 		},
 		setEmail(state, { email }) {
 			state.email = email;
+		},
+		setLoggingIn(state, value) {
+			state.loggingIn = value;
```
Comment (montyanderson) [inline ui/src/store/account.js:59]: No point passing in a value here. `setLoggingIn` should just take the state argument, then set `loggingIn` to `true`.

Comment (montyanderson) [reply ui/src/store/account.js:59]: Then set `state.loggingIn` to `false` in `setSession` :)

Comment (gabedealmeida) [reply ui/src/store/account.js:59]: @montyanderson, I see. But how about when there is a problem on the login and the section where `setSession` is called is never reached?

Comment (montyanderson) [reply ui/src/store/account.js:59]: @gabedealmeida I think this is where error handling comes in. My goal here is to make the mutations as inflexible as possible to reduce the error surface.

For me, it makes sense to have three mutations: `startLogin()`, `successLogin(email...)`, `failLogin(error)`. `isLoggingIn` should be set as a derivative of these mutations, not on it's own.

Comment (montyanderson) [reply ui/src/store/account.js:59]: Put another way, we should treat actions as effectively hostile. They have a much larger scope for failure due to concurrency, race conditions, and complexity. Mutations need to provide the most limited possible API for actions to function.

**Pull request merged**